### PR TITLE
fix: keep query redirectToPath's query params when redirecting from auth

### PR DESCRIPTION
**Pull request by alisher-aituarov**
## Summary of change

Makes redirectToPath's query params stay when redirecting to it from auth route

## Related issues

- https://github.com/supertokens/supertokens-web-js/issues/82

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [x] Changelog has been updated
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [x] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If added a new recipe interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If I added a new recipe, I also added the recipe entry point into the `size-limit` section of `package.json` with the size limit set to the current size rounded up.
-   [ ] If I added a new recipe, I also added the recipe entry point into the `rollup.config.mjs`

### CODE REVIEW

**Review by porcellus — CHANGES_REQUESTED** [review]
The changes look good, but please add tests. There are a few `redirectToPath` tests already there that you can use as a base.

File: test/end-to-end/signin.test.js [hunk 1]
```diff
@@ -486,13 +486,15 @@ describe("SuperTokens SignIn", function () {
                 `${TEST_CLIENT_BASE_URL}/redirect-here`
             );
 
-            // test that if we visit auth again, we end up in redirect-heree again
+            // test that if we visit auth again, we end up in redirect-heree again with query params kept
             await Promise.all([
-                page.goto(`${TEST_CLIENT_BASE_URL}/auth?rid=emailpassword&redirectToPath=%2Fredirect-heree`),
+                page.goto(
```
Comment (porcellus) [inline test/end-to-end/signin.test.js:491]: Please add a similar ones that test the redirection:
- emailpassword sign-in 
- emailpassword sign-up.
- passwordless sign-in
- thirdparty sign-in

This test only cheks that the auth pages redirect to the appropriate path if we navigate to it while already logged in.

File: lib/ts/utils.ts [hunk 2]
```diff
@@ -62,14 +62,19 @@ export function getURLHash(): string {
...
         return undefined;
     } else {
         // Prevent Open redirects by normalising path.
         try {
-            return new NormalisedURLPath(param).getAsStringDangerous();
+            const normalisedURLPath = new NormalisedURLPath(param).getAsStringDangerous();
+            if (includeQueryParams === false) {
+                return normalisedURLPath;
+            }
+            const pathQueryParams = param.split("?")[1] ? `?${param.split("?")[1]}` : "";
```
Comment (porcellus) [inline lib/ts/utils.ts:76]: please add a strict undefined check, so we can preserve paths ending in a `?`

**Review by porcellus — CHANGES_REQUESTED** [review]

File: lib/ts/utils.ts (hunk 2 above)
Comment (porcellus) [reply lib/ts/utils.ts:76]: This didn't happen.

File: lib/ts/utils.ts [hunk 3]
```diff
@@ -62,14 +62,19 @@ export function getURLHash(): string {
     return WindowHandlerReference.getReferenceOrThrow().windowHandler.location.getHash().substr(1);
 }
 
-export function getRedirectToPathFromURL(): string | undefined {
+export function getRedirectToPathFromURL({ includeQueryParams } = { includeQueryParams: false }): string | undefined {
```
Comment (porcellus) [inline lib/ts/utils.ts:65]: Is this used anywhere without this param set to true?

**Review by porcellus — CHANGES_REQUESTED** [review]

**Review by porcellus — CHANGES_REQUESTED** [review]
Please remove the parameter of `getRedirectToPathFromURL` if it's always used with the same parameter.

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**